File: mvpLibrary/src/main/java/cn/cbsd/mvplibrary/mvp/VDelegate.java
package cn.cbsd.mvplibrary.mvp;

import android.view.View;

/**
 * Created by wanglei on 2016/12/29.
 */

public interface VDelegate {
    void resume();

    void pause();

    void destroy();

    void showSuccess(String msg);

    void showError(String msg);

    void showError(String title, String msg);

    void showError(String title, String msg, String confirmText, View.OnClickListener listener);

    void showWarn(String msg);

    void showInfo(String msg);

    void showConfirm(String msg);

    void show(String msg);

    void showToast(String msg, int duration);

    void showLoading(String msg);

    void dismissLoading();

    //    void toastShort(String msg);
    //    void toastShort(int res);
    //    void toastLong(String msg);
    //    void toastLong(int res);

}

File: mvpLibrary/src/main/java/cn/cbsd/mvplibrary/mvp/VDelegate.kt
package cn.cbsd.mvplibrary.mvp

import android.view.View

/**
 * Created by wanglei on 2016/12/29.
 */

interface VDelegate {
    fun resume()

    fun pause()

    fun destroy()

    fun showSuccess(msg: String)

    fun showError(msg: String)

    fun showError(title: String, msg: String, confirmText: String = "我知道了", listener: View.OnClickListener? = null)

    fun showWarn(msg: String)

    fun showInfo(msg: String)

    fun showConfirm(msg: String)

    fun show(msg: String)

    fun showToast(msg: String, duration: Int)

    fun showLoading(msg: String?)

    fun dismissLoading()

    //    void toastShort(String msg);
    //    void toastShort(int res);
    //    void toastLong(String msg);
    //    void toastLong(int res);

}
